python/social_bot/tasks.py

```diff
@@ -269,16 +301,63 @@ def run(self):
                 else:
                     reward = (self._prev_dist - dist) / self._initial_dist
                 reward = reward - distraction_penalty
-                self._push_reward_queue(reward)
+                if distraction_penalty > 0:
+                    logging.debug("yielding reward: " + str(reward))
+                    self._push_reward_queue(0)
                 self._prev_dist = dist
                 agent_sentence = yield TeacherAction(
                     reward=reward, sentence=self._goal_name)
-        logging.debug("loc: " + str(loc) + " goal: " + str(goal_loc) +
-                      "dist: " + str(dist))
+        reward = -1.0
+        logging.debug("yielding reward: " + str(reward))
         self._push_reward_queue(0)
-        yield TeacherAction(reward=-1.0, sentence="failed", done=True)
+        logging.debug("reward queue len: {}, sum: {}".format(
+            str(len(self._q)), str(sum(self._q))))
+        yield TeacherAction(reward=reward, sentence="failed", done=True)
 
-    def _move_goal(self, goal, agent_loc):
+    def _get_distraction_penalty(self, agent_loc, dot, prev_min_dist_to_distraction):
+        """
+        Calculate penalty for hitting/getting close to distraction objects
+        """
+        distraction_penalty = 0
+        if (self._distraction_penalty_distance_thresh > 0 and
+            self._distraction_list):
+            curr_min_dist = 100
+            for obj_name in self._distraction_list:
+                obj = self._world.get_agent(obj_name)
+                if not obj:
+                    continue
+                obj_loc, _ = obj.get_pose()
+                obj_loc = np.array(obj_loc)
+                distraction_dist = np.linalg.norm(agent_loc - obj_loc)
+                if (distraction_dist >=
+                    self._distraction_penalty_distance_thresh):
+                    continue
+                if obj_name == self._goal_name and dot > 0.707:
```

Comment:
"dot > 0.707" means there will be a distracting penalty if closer to the goal if the angel is less than 45°, is this on purpose? Some camera horizontal view is larger than 90 degree, and in pure state observation case agent direction to the ball is not so important. Maybe we can ignore the dot here.